backend/iam/adapter.py

```diff
@@ -1,32 +1,23 @@
-from allauth.account.utils import perform_login
-from allauth.socialaccount.helpers import ImmediateHttpResponse
-from allauth.socialaccount.models import app_settings
-from django.conf import settings
+from urllib.parse import urlparse
+
 from allauth.account.adapter import DefaultAccountAdapter
 from allauth.socialaccount.adapter import (
     DefaultSocialAccountAdapter,
     MultipleObjectsReturned,
     warnings,
 )
+from allauth.socialaccount.models import app_settings
+from django.conf import settings
+from django.contrib.auth import get_user_model
 from django.db.models import Q
-from allauth.socialaccount.adapter import DefaultSocialAccountAdapter
-from allauth.core import context
-from django.dispatch import receiver
-from allauth.socialaccount.signals import pre_social_login
-from django.contrib.auth import login, get_user_model
+from django.utils.http import url_has_allowed_host_and_scheme
 from rest_framework.response import Response
 from rest_framework.status import HTTP_401_UNAUTHORIZED
-from knox.views import LoginView
-from django.utils.http import url_has_allowed_host_and_scheme
-from urllib.parse import urlparse
 
 User = get_user_model()
 
 
-class MyAccountAdapter(DefaultAccountAdapter):
-    def is_open_for_signup(self, request):
-        return False
```

Comment:
Important change is here.

This method is used by allauth to check if a user is already registered. If signup is disabled and this is set to `False`, the login flow is interrupted.

This check is redundant since we already do it in `FinishACSView`. Furthermore, in some cases, SSO for users that are already signed up fails because of this for some reason.

I took a few hours to try and debug it but it proved to be too much of a hassle to bother with it since all there is to it is to check whether a user object exists in the database.